### PR TITLE
feat: add visual feedback on API address change

### LINKED ISSUE #1635: Provide visual feedback if entered API address is up/down/invalid
This should be a quick win:

Reuse animation from  ["Add connection" on Peers screen](https://github.com/ipfs-shipyard/ipfs-webui/pull/1596#issuecomment-688807918) for newly added "API address" section on _Settings_ screen

Namely:

- red border on text entry field when value is not a valid JSON/Multiaddr/URL
- green border on text entry field when value is valid
- Display cube animation while connecting:
     - Same size and position (large, over whole screen) as is displayed when loading the Peers page
     - Same subtle "white-out" over screen content as what is displayed when using the cube as a loading indicator over the top of the "Add connection" modal
- Display feedback:
     - Red border + orange snackbar error message if failed to connect: "Could not connect to the provided API address."
     - Switch to Status page if connection is successful

### PR DESCRIPTION
**Pull request by jack-michaud**
Saw that there was a help wanted tag on #1635, thought I could lend a hand!

## Changes
- Add error text to API address settings form (`ApiAddressForm.js`)
- Populate error message for invalid addresses and connection errors (`ApiAddressForm.js`)
- Add outline to ApiAddressForm to indicate address validity (`ApiAddressForm.js`)
- Add apiAddress to "could not connect" message ("Could not connect to the IPFS API (/ip4/.../tcp/5001)") (`ipfs-provider.js`)
- Add "pending first API connection" handlers to ipfs-provider (`ipfs-provider.js`)
- Add full page loader when pending first connection (`SettingsPage.js`)

### Tried to submit an invalid IPFS API address 
The Given IPFS API Address is invalid

### Tried unsuccessfully to connect to an IPFS API
Could not connect to the IPFS API

### (Video) Filling out an API address

[Video (filling-out-address.webm)](https://ipfs.io/ipfs/QmTjkBsNoLn9BxN7uNCeF6wkddGT2hiqN1EinndgVyejF9?filename=filling-out-address.webm)

## TODO
- [x] Polish off successful state (right now it just shows no error message, but that's not nice enough)
- [x] Remove new `ApiAddressForm.css` file in favor of tachyon utility classes (I don't do anything special in the CSS) 

Closes #1635

### CODE REVIEW

**Comment by jessicaschilling** [comment]
Thanks, @jack-michaud! Assigning some folks to have a look. Please note this week is a little hectic but we'll act as speedily as we can!

**Comment by jack-michaud** [comment]
Thanks for the prompt triage @jessicaschilling ! 
Totally get that it's hectic, so no rush. I'll be chipping away at the remaining TODOs to polish it off in the meantime.

**Comment by jessicaschilling** [comment]
Hi @jack-michaud -- sounds like a plan. Drop a comment when you're ready for a final review. 😊 

I didn't dig into the code, but the video and screenshots look excellent! The only nit is that we're using snackbars for error messages (example pasted below), so for consistency's sake, probably best to present the error messages that way. (For emphasis, leaving the red border around the entry box until it's cleared by something else - like entering another address - will be helpful.)

image

(FWIW, that error was invoked by entering `/ip4/76.176.168.65/tcp/4001/p2p/QmbBHw1Xx9pUpAbrVZUKTPL5Rsph5Q9GQhRvcWVBPFgGtC` in the "Add Connection" modal on the Peers screen.)

File: src/bundles/ipfs-provider.js
Comment (rafaelramalho19) [inline src/bundles/ipfs-provider.js:340]: 💭  no need for the `{ return`, you can just use the arrow function to return directly

File: src/components/api-address-form/ApiAddressForm.js
Comment (rafaelramalho19) [inline src/components/api-address-form/ApiAddressForm.js:24]: Don't forget to use translations here

File: src/bundles/notify.js
Comment (jack-michaud) [inline src/bundles/notify.js:109]: These two errors are recycled. 
The english text for each are respectively:
- 'Could not connect. Please check if your daemon is running.'
- 'Normal IPFS service has resumed. Enjoy!'

I'd create a custom error message, but I can only responsibly source French, English, and maybe Chinese translations. Ideally, the messages would say something like:
- 'Unable to connect to the provided IPFS API address'
- 'Sucessfully connected to the IPFS API address'

Thoughts?

Comment (jessicaschilling) [reply src/bundles/notify.js:109]: Custom errors are good! We use Transifex to source internationalization and have an active network of community translators, so only need English.

**Comment by jack-michaud** [comment]
Thanks for the feedback and review @jessicaschilling and @rafaelramalho19 --
I updated the PR to use the snackbar for errors:

# Fail
image
## Success
image

I also used the internationalization keys in to eventually give this a custom translation... the `notify.json` locale file has new keys `ipfsConnectSuccess` and `ipfsConnectFail`. Are there other steps to get this into Transifex?

**Comment by jessicaschilling** [comment]
That's it for Transifex!

For the fail state, could you please give the input box a red highlight border?

**Comment by jack-michaud** [comment]
> For the fail state, could you please give the input box a red highlight border?

@jessicaschilling Absolutely! 
[Video demo](https://ipfs.io/ipfs/QmT4YTrSK2P1spXkyQVZGZZ9sL7nCQaox2HznoFXZgAq5d)

I pulled another feature from the peer "add connection" dialog behavior. For invalid multiaddrs, the submit button is disabled.
Also, like you suggested, when you submit a value and it fails to connect, the input box will remain red until new values are typed

Thank you for bearing with me in these iterations.

**Comment by jessicaschilling** [comment]
@jack-michaud This is awesome - thank you! 
@lidel and @rafaelramalho19, do you mind a review when you're able?

**Review by lidel — APPROVED** [review]
@jack-michaud reviewed functionally and this looks great, thank you! :heart:

@rafaelramalho19 please review in spare time, and merge when you feel it's ready.

File: src/bundles/notify.js
Comment (rafaelramalho19) [inline src/bundles/notify.js:73]: ⚠️ Use `===` instead (on all 3 if sentences)

**Review by rafaelramalho19 — DISMISSED** [review]
LGTM, after my change request we're good to merge 🎉

**Review by jessicaschilling — APPROVED** [review]

**Comment by jessicaschilling** [comment]
Thanks, @jack-michaud — merged!

We very much appreciate your work! If you have the inclination, please feel free to pick up any other issues labeled `help wanted` in this repo ... or for a higher-level view across all 13 GUI-related repos, check the `Quick Fixes` column of [this ZenHub board](https://app.zenhub.com/workspaces/-ipfs-app-family-ux-5e7a3123e969e659cdebb5e6/board?repos=38799513,40225364,148369983,32695583,119716282,36580101,111841602,142161410,116711586,24483721,142181521,147528357,104770273).

**Pull request merged**
